### PR TITLE
Fix: incorrect window startup position on Linux

### LINKED ISSUE #9286: Window positioned on wrong(?) monitor
**Describe the bug**
In Avalonia 0.10.18, an Avalonia window would position itself on the monitor currently being interacted with when first opened. For whatever reason, after migrating to 11.0.0-preview2, they always appear on, at least in my case, my leftmost monitor...which isn't even my primary monitor.

**To Reproduce**

1. `dotnet new avalonia.app`
2. Change version of all Avalonia `PackageReference`s to `11.0.0-preview2`
3. Add another `PackageReference` for `Avalonia.Themes.Fluent` (again, version `11.0.0-preview2`)
4. `dotnet run`

**Expected behavior**
Avalonia windows appear on the monitor currently being interacted with, instead of inexplicably ~~trying to use my reflexes to snap my neck~~ appearing all the way to the top-left of the leftmost monitor.

**Desktop:**

- OS: Linux (Kubuntu 20.04, x64)
- Version 11.0.0-preview2

**Additional context**
I'm guessing nobody else here is dumb enough to put their primary monitor/laptop on the right, besides myself? I've seen other apps suffer similar mental breakdowns whilst trying to comprehend this monitor arrangement, but *an entire GUI framework*? That could hit *every* app that uses Avalonia...

### PR DESCRIPTION
**Pull request by MinikPLayer**
## What does the pull request do?
Between release/0.10.18 and release/11.0.0-preview4 _XSizeHintsFlags.PPosition_ was added to Avalonia.X11/X11Window.cs UpdateSizeHints() flags. This resulted in incorrect window startup positioning (10px from top left corner of the leftmost screen), because system Window Managers won't auto position the window because of that hint.

## What is the current behavior?
Avalonia app window startup position is constant, and it's the top left corner of the leftmost screen.

## What is the updated/expected behavior with this PR?
Avalonia app window is auto-positioned by Window Manager (most of the time - center of the primary screen)

## How was the solution implemented (if it's not obvious)?
By removing the window Position hint - _XSizeHintsFlags.PPosition_.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Breaking changes
Couldn't find any.

## Obsoletions / Deprecations
None

## Fixed issues
Fixes #9286

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=9674) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=9674) before we can accept your contribution.<br/><br/>:x: MinikPLayer [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=9674)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=9674) it.</sub>

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027460-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-nightly/nuget/v3/index.json) [PRBUILDID]

**Comment by robloo** [comment]
Those flags were added for a reason in PR https://github.com/AvaloniaUI/Avalonia/pull/9088

> fix bug with StartupLocation on Ubuntu https://stackoverflow.com/questions/11069666/cannot-get-xcreatesimplewindow-to-open-window-at-the-right-position
>
> On some linux-distros windows managers ignors position and size of you window (set throught XCreateSimpleWindow) until you tell him to consider your settings.

I'm not a Linux dev at all I just remember the last PR so thought I would mention it. Whatever is decided, this needs to be documented **IN CODE** so people don't forget and flip things back and forth like this in the future.

@dif-sam (who made the last change) @kekekeks (approved)

**Comment by kekekeks** [comment]
Try setting `WindowStartupLocation="CenterScreen"`, since by default it's `Manual`

We should probably consider adding `WindowStartupLocation.Auto`  and unset PPosition in that case to let the window manager to do its thing.

**Comment by MinikPLayer** [comment]
You're right, there are some situations in which these flags are required, so I've created a new commit introducing changes to better reflect other platforms behaviour.
Now if WindowStartupLocation is set to "Manual" / not set and Position parameter is **not** set, then Window is auto-positioned by Window Manager. If Position is set, then window sets it's position to the parameter value.
Other WindowStartupLocations work as intended.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027474-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-nightly/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027508-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027591-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Avalonia.X11/X11Window.cs [hunk 1]
```diff
@@ -958,6 +962,9 @@ public PixelPoint Position
             get => _position ?? default;
             set
             {
+                _usePositioningFlags = true;
+                UpdateSizeHints(null);
```
[the remark this answers is not in the document]
Comment (MinikPLayer) [reply src/Avalonia.X11/X11Window.cs]: Sure, you're right. Changed / fixed in the newest commit a5e018f77957d50f6ace60b034f7bf24c0092af0

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027706-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0027723-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

**Pull request merged**